### DISCUSSION TITLE
请问图片的防盗链功能是怎么实现的？

### DISCUSSION THREAD

**Discussion by a1270107629** [yangrunkang/upupor#60] [Q&A] [answered]
在学习这个项目的时候，看了很长时间的代码，这个问题一直没搞懂

**Comment by yangrunkang** [comment]
> 在学习这个项目的时候，看了很长时间的代码，这个问题一直没搞懂

图片存储使用的是阿里的oss,在阿里的oss可以设置防盗链。如果自己实现防盗链可以在nginx实现，不是在项目中实现
<img width="1367" alt="image" src="https://user-images.githubusercontent.com/46257933/168030151-4cfd5af6-6102-4681-b47a-82a91ce57340.png">


> 接入oss的通知 https://www.upupor.com/u/20041214131111593984

**Comment by yangrunkang** [comment, marked as answer]
@a1270107629 目前已经弃用阿里OSS了,基于minio自建实现了
